src/utils/DependencyUtils.ts

```diff
@@ -11,3 +29,71 @@ export const splitToNameAndVersion = (dependencyString: string): [string, string
     const fullName = parts.join('-');
     return [fullName, version];
 };
+
+/**
+ * - When installing a regular mods, always install the latest versions of dependencies.
+ * - When installing a mod pack we usually want to ensure compatibility by installing the exact
+ *      versions of dependencies, even if it means upgrading/downgrading already installed mods.
+ *   - If, however, modpackDependencyStrategy is set to USE_LATEST_VERSION, we install the latest
+ *      versions of dependencies for mod packs too.
+ * - When installing latest versions of dependencies, already installed mods are filtered out.
+ * - The order of the input combos array matters, combos earlier on the array get their dependencies'
+ *      versions prioritised over later ones.
+ */
+export async function getFullDependencyList(
+    combos: ThunderstoreCombo[],
+    game: Game,
+    installedMods: ManifestV2[],
+    modpackDependencyStrategy: ModpackDependencyStrategy
+) {
+    const originalCombos = [...combos];
```

Comment:
I think there's no need for this method to mutate the `combos` argument, it can as well return a new array which makes it easier to grasp what's going on in the call site. (And functions generally should have side effects only if there's a good reason for it).